src/lsm/table_memory.zig

```diff
@@ -196,20 +218,84 @@ pub fn TableMemoryType(comptime Table: type) type {
             } };
         }
 
+        // Check if incremental sorted runs are strictly monotonic increasing.
+        // If that is the case, we can skip the final sort.
+        fn sorted_runs_monotonic(table: *const TableMemory) bool {
+            // No sorted runs or the last sorted run does not cover the full table.
+            // This can happen with scans.
+            if (table.sorted_runs_count == 0) return false;
+            if (table.sorted_runs[table.sorted_runs_count - 1].?.max != table.count()) {
+                return false;
+            }
+
+            var skip_final_sort = true;
+
+            for (
+                table.sorted_runs[0..(table.sorted_runs_count - 1)],
+                table.sorted_runs[1..table.sorted_runs_count],
+            ) |maybe_sorted_run_a, maybe_sorted_run_b| {
+                const sorted_run_a = maybe_sorted_run_a orelse continue;
```

Comment:
these `continue` are actually `unreachable`, right? Let's do this more directly, by not making `SortedRun`s optional, and instead initializing with `min = max = maxInt(u32)`